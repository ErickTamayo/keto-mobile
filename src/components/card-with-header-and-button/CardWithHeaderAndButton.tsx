import React, { ReactNode } from 'react'
import { Text, View, GestureResponderEvent } from 'react-native'
import Button from '../button/Button'
import {
  containerStyle,
  cardContainerStyle,
  cardStyle,
  titleStyle,
  descriptionStyle,
  childrenContainerStyle,
  textContainerStyle,
  footerStyle,
} from './styles'

export interface Props {
  title: string
  description?: string
  buttonText: string
  onButtonPress?: (event: GestureResponderEvent) => void
  disabled?: boolean
  loading?: boolean
  children?: ReactNode
  footer?: ReactNode
}

const CardWithHeaderAndButton = ({
  title,
  description,
  buttonText,
  onButtonPress,
  disabled,
  children,
  footer,
}: Props) => (
  <View style={containerStyle}>
    <View style={textContainerStyle}>
      <Text style={titleStyle}>{title}</Text>
      <Text style={descriptionStyle}>{description}</Text>
    </View>
    <View style={cardContainerStyle}>
      <View style={cardStyle}>
        <View style={childrenContainerStyle}>{children}</View>
        <Button
          title={buttonText}
          variant="primary-md"
          onPress={onButtonPress}
          disabled={disabled}
        />
      </View>
      {footer && <View style={footerStyle}>{footer}</View>}
    </View>
  </View>
)

export default CardWithHeaderAndButton
