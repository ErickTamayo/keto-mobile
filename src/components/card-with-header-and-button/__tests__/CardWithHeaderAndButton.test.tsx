import 'react-native'
import * as React from 'react'
import { Text, TouchableOpacity, GestureResponderEvent } from 'react-native'
import { shallow, mount } from 'enzyme'
import CardWithHeaderAndButton, { Props } from '../CardWithHeaderAndButton'

describe('<CardWithHeaderAndButton />', () => {
  it('renders correctly', () => {
    const props: Props = {
      title: 'card-title',
      buttonText: 'button-text',
    }
    const wrapper = shallow(
      <CardWithHeaderAndButton {...props}>
        <Text>card-children</Text>
      </CardWithHeaderAndButton>
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('renders disabled', () => {
    const props: Props = {
      title: 'card-title',
      buttonText: 'button-text',
      disabled: true,
    }
    const wrapper = shallow(
      <CardWithHeaderAndButton {...props}>
        <Text>card-children</Text>
      </CardWithHeaderAndButton>
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('should trigger onPress once pressed', () => {
    const props: Props = {
      title: 'card-title',
      buttonText: 'button-text',
      onButtonPress: jest.fn(),
    }
    const wrapper = mount(<CardWithHeaderAndButton {...props} />)
    wrapper.find(TouchableOpacity).prop('onPress')!({} as GestureResponderEvent)

    expect(props.onButtonPress).toBeCalled()
  })
})
