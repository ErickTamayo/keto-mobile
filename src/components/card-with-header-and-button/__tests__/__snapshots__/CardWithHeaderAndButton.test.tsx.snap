// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardWithHeaderAndButton /> renders correctly 1`] = `
<Component
  style={
    Array [
      Object {
        "flex": 1,
      },
      Object {
        "justifyContent": "space-around",
      },
    ]
  }
>
  <Component
    style={
      Array [
        Object {
          "flex": 1,
        },
        Object {
          "alignItems": "center",
        },
        Object {
          "justifyContent": "center",
        },
      ]
    }
  >
    <Component
      style={
        Array [
          Object {
            "fontFamily": "avenir-next-medium",
          },
          Object {
            "color": "#626262",
          },
          Object {
            "textAlign": "center",
          },
          Object {
            "fontSize": 20,
          },
          Object {
            "margin": 19.5,
          },
          Object {
            "marginBottom": 32.5,
          },
        ]
      }
    >
      card-title
    </Component>
    <Component
      style={
        Array [
          Object {
            "fontFamily": "avenir-next-regular",
          },
          Object {
            "color": "#9e9e9e",
          },
          Object {
            "fontSize": 16,
          },
          Object {
            "textAlign": "center",
          },
          Object {
            "marginLeft": 19.5,
          },
          Object {
            "marginRight": 26,
          },
        ]
      }
    />
  </Component>
  <Component
    style={
      Array [
        Object {
          "flex": 2,
        },
        Object {
          "alignItems": "center",
        },
      ]
    }
  >
    <Component
      style={
        Array [
          Object {
            "marginTop": 13,
          },
          Object {
            "paddingLeft": 19.5,
          },
          Object {
            "paddingRight": 19.5,
          },
          Object {
            "paddingBottom": 26,
          },
          Object {
            "paddingTop": 26,
          },
          Object {
            "backgroundColor": "#ffffff",
          },
          Object {
            "alignItems": "center",
          },
          Object {
            "justifyContent": "center",
          },
          Object {
            "borderRadius": 10,
          },
          Object {
            "shadowColor": "#000000",
            "shadowOffset": Object {
              "height": 4,
              "width": 2,
            },
            "shadowOpacity": 0.1,
            "shadowRadius": 2,
          },
        ]
      }
    >
      <Component
        style={Array []}
      >
        <Component>
          card-children
        </Component>
      </Component>
      <Button
        title="button-text"
        variant="primary-md"
      />
    </Component>
  </Component>
</Component>
`;

exports[`<CardWithHeaderAndButton /> renders disabled 1`] = `
<Component
  style={
    Array [
      Object {
        "flex": 1,
      },
      Object {
        "justifyContent": "space-around",
      },
    ]
  }
>
  <Component
    style={
      Array [
        Object {
          "flex": 1,
        },
        Object {
          "alignItems": "center",
        },
        Object {
          "justifyContent": "center",
        },
      ]
    }
  >
    <Component
      style={
        Array [
          Object {
            "fontFamily": "avenir-next-medium",
          },
          Object {
            "color": "#626262",
          },
          Object {
            "textAlign": "center",
          },
          Object {
            "fontSize": 20,
          },
          Object {
            "margin": 19.5,
          },
          Object {
            "marginBottom": 32.5,
          },
        ]
      }
    >
      card-title
    </Component>
    <Component
      style={
        Array [
          Object {
            "fontFamily": "avenir-next-regular",
          },
          Object {
            "color": "#9e9e9e",
          },
          Object {
            "fontSize": 16,
          },
          Object {
            "textAlign": "center",
          },
          Object {
            "marginLeft": 19.5,
          },
          Object {
            "marginRight": 26,
          },
        ]
      }
    />
  </Component>
  <Component
    style={
      Array [
        Object {
          "flex": 2,
        },
        Object {
          "alignItems": "center",
        },
      ]
    }
  >
    <Component
      style={
        Array [
          Object {
            "marginTop": 13,
          },
          Object {
            "paddingLeft": 19.5,
          },
          Object {
            "paddingRight": 19.5,
          },
          Object {
            "paddingBottom": 26,
          },
          Object {
            "paddingTop": 26,
          },
          Object {
            "backgroundColor": "#ffffff",
          },
          Object {
            "alignItems": "center",
          },
          Object {
            "justifyContent": "center",
          },
          Object {
            "borderRadius": 10,
          },
          Object {
            "shadowColor": "#000000",
            "shadowOffset": Object {
              "height": 4,
              "width": 2,
            },
            "shadowOpacity": 0.1,
            "shadowRadius": 2,
          },
        ]
      }
    >
      <Component
        style={Array []}
      >
        <Component>
          card-children
        </Component>
      </Component>
      <Button
        disabled={true}
        title="button-text"
        variant="primary-md"
      />
    </Component>
  </Component>
</Component>
`;
